modal/_container_entrypoint.py

```diff
@@ -858,28 +865,41 @@ def import_function(
     else:
         obj = None
 
-    if not pty_info.pty_type:  # do not wrap PTY-enabled functions
+    if function_def.webhook_config.type:
+        is_async = True
+        is_generator = True
+        data_format = api_pb2.DATA_FORMAT_ASGI
+
         if function_def.webhook_config.type == api_pb2.WEBHOOK_TYPE_ASGI_APP:
-            # function returns an asgi_app, that we can use as a callable.
-            asgi_app = fun()
-            fun = asgi_app_wrapper(asgi_app, function_io_manager)
-            is_async = True
-            is_generator = True
-            data_format = api_pb2.DATA_FORMAT_ASGI
+            # Function returns an asgi_app, which we can use as a callable.
+            fun = asgi_app_wrapper(fun(), function_io_manager)
+
         elif function_def.webhook_config.type == api_pb2.WEBHOOK_TYPE_WSGI_APP:
-            # function returns an wsgi_app, that we can use as a callable.
-            wsgi_app = fun()
-            fun = wsgi_app_wrapper(wsgi_app, function_io_manager)
-            is_async = True
-            is_generator = True
-            data_format = api_pb2.DATA_FORMAT_ASGI
+            # Function returns an wsgi_app, which we can use as a callable.
+            fun = wsgi_app_wrapper(fun(), function_io_manager)
+
         elif function_def.webhook_config.type == api_pb2.WEBHOOK_TYPE_FUNCTION:
-            # function is webhook without an ASGI app. Create one for it.
-            asgi_app = webhook_asgi_app(fun, function_def.webhook_config.method)
-            fun = asgi_app_wrapper(asgi_app, function_io_manager)
-            is_async = True
-            is_generator = True
-            data_format = api_pb2.DATA_FORMAT_ASGI
+            # Function is a webhook without an ASGI app. Create one for it.
+            fun = asgi_app_wrapper(
+                webhook_asgi_app(fun, function_def.webhook_config.method),
+                function_io_manager,
+            )
+
+        elif function_def.webhook_config.type == api_pb2.WEBHOOK_TYPE_WEB_SERVER:
+            # Function spawns an HTTP web server listening at a port.
+            fun()
```

Comment:
It might not be obvious for users that the function has to return after spawning a background server - perhaps we could run `fun` in a thread just to avoid the foot gun of making a blocking call to something that runs a server?

Reply:
Makes sense, I think we'll stick with this for now just because moving it to a thread is less likely to break users in the future, and also they can run `threading.Thread()` themselves if needed.